### PR TITLE
Closes #1005: Manage user authentication state using React Context

### LINKED ISSUE #1005: Manage user authenticaion state using React Context
We need to improve how we deal with logged in user state in the frontend app.  Currently we determine that a user is logged in using the return code (403 or 200) of hitting the `/user/info` route.  If a user is logged in, we give back a User Object, for example:

```json
{
  "name": "David Humphrey",
  "email": "david.humphrey@sencollege.ca",
  "id": "5cb1175813",
  "isAdmin": true
}
```

This information affects how a bunch of things work in the frontend.  Because lots of different components need it, I'm suggesting that we explore using the [React Context](https://reactjs.org/docs/context.html) to share the User object through our pages and component tree without having to use props. 

This [blog](https://kentcdodds.com/blog/authentication-in-react-applications) might help as well.

Once we do this, we can start refactoring the frontend to use the User context and remove localized hits to `/user/info` and props.

### PR DESCRIPTION
**Pull request by manekenpix**
Co-authored-by: c3ho <calvinho125@gmail.com>

## Issue This PR Addresses

Closes #1005 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This adds [React Context](https://reactjs.org/docs/context.html) to share the `User` object through our pages and component tree without having to use props.
Because of how our workflow uses redirection for authentication, this PR needs to be tested fully locally, and it probably won't work on `Vercel`.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by c3ho — APPROVED** [review]
Was able to successfully login using user1/user1pass. Successfully visited pages and had the correct privileges such as adding/deleting feeds.

File: src/frontend/src/components/Login/LoggedIn.jsx [hunk 1]
```diff
@@ -29,7 +30,8 @@ const useStyles = makeStyles({
   },
 });
 
-function LoggedIn(props) {
+function LoggedIn() {
+  const state = useContext(UserStateContext);
```
Comment (humphd) [inline src/frontend/src/components/Login/LoggedIn.jsx:34]: s/state/user/ would be more clear, I think.

File: src/frontend/src/components/Login/Login.jsx [hunk 2]
```diff
@@ -10,12 +11,13 @@ import useSiteMetadata from '../../hooks/use-site-metadata';
 
 function Login() {
   const { telescopeUrl } = useSiteMetadata();
-  const [email, setEmail] = useState(null);
-  const [name, setName] = useState(null);
+  const state = useContext(UserStateContext);
```
Comment (humphd) [inline src/frontend/src/components/Login/Login.jsx:14]: Same thing here, I think `user` is better.

File: src/frontend/src/pages/myfeeds.js [hunk 3]
```diff
@@ -21,44 +22,24 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 export default function MyFeeds() {
+  const state = useContext(UserStateContext);
```
Comment (humphd) [inline src/frontend/src/pages/myfeeds.js:25]: `user`

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Review by c3ho — APPROVED** [review]

**Pull request merged**